pkg/plugins/envvars/envvars.go

```diff
@@ -180,6 +180,13 @@ func (s *Service) tracingEnvVars(plugin *plugins.Plugin) []string {
 		fmt.Sprintf("GF_INSTANCE_OTLP_ADDRESS=%s", s.cfg.Tracing.OpenTelemetry.Address),
 		fmt.Sprintf("GF_INSTANCE_OTLP_PROPAGATION=%s", s.cfg.Tracing.OpenTelemetry.Propagation),
 	}
+	if s.cfg.Tracing.OpenTelemetry.Sampler != "" || s.cfg.Tracing.OpenTelemetry.SamplerParam != 0 {
```

Comment:
Might be worth adding a comment here maybe to explain the usage since it seems only 1 actually needs to be set for this to be used, while `SamplerRemoteURL` can always be empty for example.

Reply:
I have added comments and changed it so it passes the remote sampler url only if it's set, it should be less confusing this way :smiley:

Reply:
To confirm, it makes sense to set `GF_INSTANCE_OTLP_SAMPLER_REMOTE_URL` on its own (in the case where the others aren't)? IE the `s.cfg.Tracing.OpenTelemetry.SamplerRemoteURL` check should live outside the first check

Reply:
More or less, but after checking again, I concluded can just always pass those env vars to the process because `pkg/infra/tracing` will take caring of the default values (like `samplerParam` = 1.0), and we can just pass those value to the plugin:

https://github.com/grafana/grafana/blob/f7361a27553acbba399e00e948a4241873c0cff9/pkg/infra/tracing/tracing.go#L146-L148

https://github.com/grafana/grafana/blob/705655d3a931285d5a06745e16ebff52ab2d5c60/pkg/services/pluginsintegration/config/tracing.go#L25-L27

Changes: https://github.com/grafana/grafana/pull/76834/commits/ccc50caa9c5041016630cb941743281c74c1b661

This complicated the test a bit but simplifies the code